src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -170,17 +167,25 @@ public override PSCredential PromptForCredential(string caption, string message,
 
         private static bool SecureStringEquals(SecureString password, SecureString confirmPassword)
         {
+            if(password.Length != confirmPassword.Length)
+            {
+                return false;
+            }
+
+            int length = 2 * password.Length;
             IntPtr pwd_ptr = IntPtr.Zero;
             IntPtr confirmPwd_ptr = IntPtr.Zero;
+            byte[] pwd = new byte[length];
+            byte[] confirmPwd = new byte[length];
             try
             {
                 pwd_ptr = Marshal.SecureStringToBSTR(password);
-                string pwd = Marshal.PtrToStringBSTR(pwd_ptr);
+                Marshal.Copy(pwd_ptr, pwd, 0, length);
```

Comment:
We have no need to make the copy. See https://www.sjoerdlangkemper.nl/2017/11/08/comparing-securestrings-in-dotnet/
```c#
static bool IsEqual(IntPtr bstr1, IntPtr bstr2)
{
    var length1 = Marshal.ReadInt32(bstr1, -4);
    var length2 = Marshal.ReadInt32(bstr2, -4);

    if (length1 != length2) return false;

    var equal = 0;
    for (var i = 0; i < length1; i++)
    {
        var c1 = Marshal.ReadByte(bstr1 + i);
        var c2 = Marshal.ReadByte(bstr2 + i);
        equal |= c1 ^ c2;
    }
    return equal == 0;
}
```

Reply:
@iSazonov I have made the changes. Please review and let me know your feedback.